tools/archive_metadata_validator/utils/Rx.py

```diff
@@ -0,0 +1,590 @@
+import re
+from six import string_types  # for 2-3 compatibility
```

Comment:
FTR: From the datalad POV we can safely ignore PY2 and develop PY3 only.

Reply:
This file seems to be a copy from https://github.com/rjbs/Rx/blob/master/python/Rx.py

Should be mentioned in the header or copy avoided at all?

Reply:
Also what license and copyright

Reply:
Yes, you are right. It should be mentioned (it was only mentioned in schema definition). I will fix this.

The license is GPL which should be OK for us unless we want a more liberal license, e.g. MIT, to be used.

I picked Rx for the first attempt to have some schema validation. It can also be replaced with another tool if you have one in mind.

Reply:
BTW, might of interest for the future: https://pydantic-docs.helpmanual.io/ to establish schema (and validation and export) based on Python data types description